cmd/iq.go

```diff
@@ -0,0 +1,164 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sonatype-nexus-community/ahab/buildversion"
+	"github.com/sonatype-nexus-community/ahab/packages"
+	"github.com/sonatype-nexus-community/ahab/parse"
+	"github.com/sonatype-nexus-community/go-sona-types/iq"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var (
+	iqUsername  string
+	iqToken     string
+	iqHost      string
+	stage       string
+	application string
+	maxRetries  int
+	lifecycle   *iq.Server
+)
+
+var iqCmd = &cobra.Command{
+	Use:   "iq",
+	Short: "iq is used for auditing your projects with Nexus IQ Server",
+	RunE: func(cmd *cobra.Command, args []string) (err error) {
+		fflags := cmd.Flags()
+
+		err = checkRequiredFlags(fflags)
+		if err != nil {
+			return
+		}
+
+		printHeader()
+		logger = logrus.New()
+		switch verbose {
+		case 1:
+			logger.Level = logrus.InfoLevel
+		case 2:
+			logger.Level = logrus.DebugLevel
+		case 3:
+			logger.Level = logrus.TraceLevel
+		default:
+			logger.Level = logrus.ErrorLevel
+		}
+
+		lifecycle = iq.New(logger,
+			iq.Options{
+				User:          iqUsername,
+				Token:         iqToken,
+				Server:        iqHost,
+				Application:   application,
+				Stage:         stage,
+				Tool:          "ahab-client",
+				Version:       buildversion.BuildVersion,
+				OSSIndexToken: ossIndexToken,
+				OSSIndexUser:  ossIndexUser,
+				DBCacheName:   "ahab-cache",
+				MaxRetries:    maxRetries,
+			})
+
+		var list []string
+		scanner := bufio.NewScanner(os.Stdin)
+		for scanner.Scan() {
+			list = append(list, scanner.Text())
+		}
+		if err = scanner.Err(); err != nil {
+			if err != nil {
+				return
+			}
+		}
+
+		res, err := lifecycle.AuditPackages(tryParseStdInListIQ(list, &operating), application)
+		if err != nil {
+			return
+		}
+
+		fmt.Println()
+		if res.IsError {
+			logger.WithField("res", res).Error("An error occurred with the request to IQ Server")
+			return fmt.Errorf("Uh oh! There was an error with your request to Nexus IQ Server")
+		}
+
+		if res.PolicyAction != "Failure" {
+			logger.WithField("res", res).Debug("Successful in communicating with IQ Server")
+			fmt.Println("Wonderbar! No policy violations reported for this audit!")
+			fmt.Println("Report URL: ", res.ReportHTMLURL)
+			os.Exit(0)
+		} else {
+			logger.WithField("res", res).Debug("Successful in communicating with IQ Server")
+			fmt.Println("Hi, Nancy here, you have some policy violations to clean up!")
```

Comment:
Copy pasta... should probably be `Hi, Ahab here.....`

Reply:
In: 1f4f90f